### PR TITLE
engine: fix string to numeric conversions

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR fixes `FROM_STRING_NUMERIC`  implementation in the context of Numeric implementation #2289.

In particular it makes the builtin support huge string without taking too much time or blowing.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [X] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala [hunk 1]
```diff
@@ -984,20 +1001,20 @@ class SBuiltinTest extends FreeSpec with Matchers with TableDrivenPropertyChecks
...
-          "pi",
-          "0x11",
-          "1E10",
-          "2.",
-          "1L",
-          "+-1",
-          "1" * 29,
-          "-" + "1" * 29,
-          "+" + "1" * 29,
+          //"pi",
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala]: Uncomment again?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 2]
```diff
@@ -292,15 +292,37 @@ object SBuiltin {
...
       val scale = args.get(0).asInstanceOf[STNat].n
       val string = args.get(1).asInstanceOf[SText].value
-      val num = Numeric
-        .fromUnscaledString(string)
-        .flatMap(Numeric.fromBigDecimal(scale, _))
-        .fold(_ => None, x => Some(SNumeric(x)))
-      machine.ctrl = CtrlValue(SOptional(num))
+      val result = string match {
+        case validFormat(signPart, intPart, _, decPartOrNull) =>
+          val decPart = Option(decPartOrNull).filterNot(_ == "0").getOrElse("")
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: Doesn't this filter out all `0` characters of the decimal part, also those in `1.0001`?

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: `filter` is called on  `Option[String]` not in `String`.

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala]: fixed

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 2 above)
Comment (gerolf-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: D'oh, of course. Thank you.

**Pull request merged**